### ISSUE TITLE
bower install not working

### ISSUE THREAD

**Issue by semal** [OnroerendErfgoed/atramhasis#461]
When I run "bower install" at admin folder, no dist folder and src folder generated. but the folders is required when I using the web UI to edit SKOS objects. Please help..

**Comment by koenedaele** [comment]
@cedrikv Is bower still functional?

**Comment by cedrikv** [comment]
> @cedrikv Is bower still functional?

Yes, we don't have any issues with Bower in other projects.

> When I run "bower install" at admin folder, no dist folder and src folder generated.

The src folder is already provided in the code:
https://github.com/OnroerendErfgoed/atramhasis/tree/master/atramhasis/static/admin/src

The dist folder is not necessary to test the app, you need it to run in production mode. It is created after compiling the dojo code with this command in the admin folder:
`grunt -v build` 

 Bower installs all the external javascript libraries you need to run the project, like Dojo. After running 'bower install', you should see a lot of extra folders in the 'admin/src' directory, like 'dojo'.

**Comment by Jefwillems** [comment]
I'm running into the same issue but the problem is not related to bower install. 

When scaffolding a project with:
`pcreate -s atramhasis_scaffold my_thesaurus`

the folders `src`, `profiles` and the file `.bowerrc` are not generated.

also grunt is missing as a dependency so `grunt -v build` won't run in docker builds. 

@semal As a temporary workaround you can let docker (or yourself) copy the admin folder into the container, this seems to work.

**Comment by Jefwillems** [comment]
@semal have a look at [this repository](https://github.com/lab9k/atramhasis_docker) for a working docker example

**Comment by koenedaele** [comment]
There's also https://github.com/OnroerendErfgoed/atramhasis-demo-docker for a demo version using docker.

**Comment by koenedaele** [comment]
I would like to understand what's going on here. @semal could you link to the docs you've been using, the steps you've taken and what does and doesn't work. Then I can try to reproduce.

**Comment by koenedaele** [comment]
Steps I've taken in reproducing this:

- I followed the instructions in https://atramhasis.readthedocs.io/en/latest/customisation.html#creating-your-own-project (creating your own project)
- First issue: circular python deps. I fixed this by editing atramhasis-requirements-dev.txt and commenting the second line (this is an error and should be fixed)
- When I get to the bower install step, I get the following error:

```bash
[koen@Coraline:~/Projecten/python/skos/koenthes/koenthes/static]
[19:41:02] $ bower install
bower jquery#~2.1.4            EINVRES Request to https://bower.herokuapp.com/packages/jquery failed with 502
```

If I run this command several times, it sometimes complains about jquery, sometimes about foundation.

But, if I follow these instructions: https://stackoverflow.com/questions/51020317/einvres-request-to-https-bower-herokuapp-com-packages-failed-with-502 and update bower to the latest version, the bower install works.

@cedrikv I also do not get the src folders when following the instructions, so something's wrong somewhere.

**Comment by koenedaele** [comment]
It looks like the last version (0.6.5) has packaging errors and does not include the src folders. This can be see on pypi since the 0.6.5 version is only 2.4MB, while the 0.6.4 version is 26.6MB.

@goessebr I think you release the last version and forgot to run the "prepare" command. See https://atramhasis.readthedocs.io/en/latest/development.html#distribution

**Comment by koenedaele** [comment]
@semal @Jefwillems Workaround would be to use the 0.6.4 version until we release an updated version.

**Comment by Jefwillems** [comment]
@koenedaele That's the docker image i based mine off, while fixing the bugs it has. Thanks for the quick help with the versions. i'll downgrade today

**Comment by koenedaele** [comment]
What bugs did that docker image have? I didn't write it myself, but if it also tries to install 0.6.5 it won't work either.

**Comment by Jefwillems** [comment]
I had to specify the pip version with pip==9.0.3 , because it would throw errors otherwise. 
I also had to add java because it's a dependency of the dojo optimizer.

**Comment by cedrikv** [comment]
fixed in version 0.6.7
